keras/applications/nasnet.py

```diff
@@ -0,0 +1,692 @@
+for ImageNet 2012 are provided.
+
+The below table describes the performance on ImageNet 2012:
+------------------------------------------------------------------------------------
+      Architecture       | Top-1 Acc | Top-5 Acc |  Multiply-Adds |  Params (M)
+------------------------------------------------------------------------------------
+|   NASNet-A (4 @ 1056)  |   74.0 %  |   91.6 %  |       564 M    |     5.3        |
+|   NASNet-A (6 @ 4032)  |   82.7 %  |   96.2 %  |      23.8 B    |    88.9        |
+------------------------------------------------------------------------------------
+
+Weights obtained from the official Tensorflow repository found at
+https://github.com/tensorflow/models/tree/master/research/slim/nets/nasnet
+
+# References:
+ - [Learning Transferable Architectures for Scalable Image Recognition]
+    (https://arxiv.org/abs/1707.07012)
+
+Based on the following implementations:
+ - [TF Slim Implementation]
+   (https://github.com/tensorflow/models/blob/master/research/slim/nets/nasnet/nasnet.py)
+ - [TensorNets implementation]
+   (https://github.com/taehoonlee/tensornets/blob/master/tensornets/nasnets.py)
+"""
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+import warnings
+
+from ..models import Model
+from ..layers import Input
+from ..layers import Activation
+from ..layers import Dense
+from ..layers import Dropout
+from ..layers import BatchNormalization
+from ..layers import MaxPooling2D
+from ..layers import AveragePooling2D
+from ..layers import GlobalAveragePooling2D
+from ..layers import GlobalMaxPooling2D
+from ..layers import Conv2D
+from ..layers import SeparableConv2D
+from ..layers import ZeroPadding2D
+from ..layers import Cropping2D
+from ..layers import concatenate
+from ..layers import add
+from ..regularizers import l2
+from ..utils.data_utils import get_file
+from ..engine.topology import get_source_inputs
+from ..applications.imagenet_utils import _obtain_input_shape
+from ..applications.inception_v3 import preprocess_input
+from ..applications.imagenet_utils import decode_predictions
+from .. import backend as K
+
+_BN_DECAY = 0.9997
+_BN_EPSILON = 1e-3
+
+NASNET_MOBILE_WEIGHT_PATH = "https://github.com/titu1994/Keras-NASNet/releases/download/v1.0/NASNet-mobile.h5"
+NASNET_MOBILE_WEIGHT_PATH_NO_TOP = "https://github.com/titu1994/Keras-NASNet/releases/download/v1.0/NASNet-mobile-no-top.h5"
+NASNET_LARGE_WEIGHT_PATH = "https://github.com/titu1994/Keras-NASNet/releases/download/v1.1/NASNet-large.h5"
+NASNET_LARGE_WEIGHT_PATH_NO_TOP = "https://github.com/titu1994/Keras-NASNet/releases/download/v1.1/NASNet-large-no-top.h5"
+
+
+def NASNet(input_shape=None,
+           penultimate_filters=4032,
+           nb_blocks=6,
+           stem_filters=96,
+           skip_reduction=True,
+           filters_multiplier=2,
+           dropout=0.5,
+           weight_decay=5e-5,
```

Comment:
Dropout, weight decay are regularization parameter and should not be included.

Reply:
I would argue that for fine tuning purposes (with `include_top=False`), weight decay is important. 

As to dropout, I included it since the original codebase uses it with these values. I can remove them from the Keras version, but the weights will need to be updated on your end to ignore the Dropout layer (or use by_name=True)